pgjdbc/src/test/java/org/postgresql/test/jdbc4/Binary2StringTest.java

```diff
@@ -0,0 +1,55 @@
+package org.postgresql.test.jdbc4;
+
+import static org.junit.Assert.assertNotNull;
+
+import org.postgresql.PGStatement;
+import org.postgresql.jdbc.PreferQueryMode;
+import org.postgresql.test.TestUtil;
+import org.postgresql.test.jdbc2.BaseTest4;
+
+import org.junit.After;
+import org.junit.Assume;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+
+public class Binary2StringTest extends BaseTest4 {
+
+  private PreparedStatement statement;
+
+  @Before
+  public void setUp() throws Exception {
+    super.setUp();
+    Assume.assumeTrue("Server-prepared statements are not supported in 'simple protocol only'",
+        preferQueryMode != PreferQueryMode.EXTENDED_FOR_PREPARED);
+    TestUtil.createTable(con, "testGeometric", "p point");
+    Statement stmt = con.createStatement();
+    stmt.execute("insert into testGeometric(p) values ('(1,1)'),('(2,2)')");
+    statement = con.prepareStatement("select p from testGeometric");
+
+    ((PGStatement) statement).setPrepareThreshold(-1);
```

Comment:
I've started to review this, however I ended up with augmenting the existing test: https://github.com/pgjdbc/pgjdbc/pull/1155

Note: there was a test for bunch of geometric types. The missing bits are `binary/text` and `getString`.